### PR TITLE
switch to composite build and add api key check

### PR DESCRIPTION
**Pull request by willnorris**
This prepares for merging #24.  Manually tested in a private repo on a test tailnet.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -15,11 +15,24 @@ inputs:
...
+  steps:
+    - name: Check Auth Info Empty
+      if: ${{ inputs.api-key == '' }}
+      shell: bash
+      run: |
+        echo "::error title=⛔ error hint::API Key empty. Maybe you need to populate it in the Secrets for your workflow, see more in https://docs.github.com/en/actions/security-guides/encrypted-secrets"
+        exit 1
+    - uses: actions/setup-go@v4.0.0
+      with:
+        go-version: 1.21.6
```
Comment (DentonGentry) [inline action.yml:28]: I'd recommend referencing the go.mod file so as to not have to increment this manually:
```
     - name: Install Go
       uses: actions/setup-go@v4
       with:
         go-version-file: go.mod
         cache: false
```

**Review by DentonGentry — APPROVED** [review]

File: action.yml (hunk 1 above)
Comment (willnorris) [reply action.yml:28]: There is no go.mod file in this repo, and I don't think we can reference the one from tailscale/tailscale

Comment (DentonGentry) [reply action.yml:28]: Oh this isn't OSS. There isn't a go.mod. Never mind.

**Comment by oscar-b** [comment]
@willnorris Hey! This seems to break on our self hosted runners:

```
Run go install tailscale.com/cmd/gitops-pusher@gitops-1.58.2
  go install tailscale.com/cmd/gitops-pusher@gitops-1.58.2
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    GOBIN: /usr/local/bin/
tailscale.com/cmd/gitops-pusher: go install tailscale.com/cmd/gitops-pusher: copying /tmp/go-build2096326023/b001/exe/a.out: open /usr/local/bin/gitops-pusher: permission denied
```

Not sure if we should just allow our gh user access to `/usr/local/bin`? Not sure how this is configured on GitHubs runners.

**Comment by oscar-b** [comment]
As an alternative solution, I'm running the workflow in a container instead:

```
container: golang:1.21.6-bookworm
``

**Pull request merged**